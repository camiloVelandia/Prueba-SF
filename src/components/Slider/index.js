import React, { useState, useEffect, useContext } from "react";
import AppContext from "../../context/AppContext";

// import Swiper core and required modules
import SwiperCore, { Navigation, Pagination, Scrollbar, A11y } from 'swiper';
import { Swiper, SwiperSlide } from 'swiper/react';
// Import Swiper styles
import "swiper/swiper-bundle.min.css";

// Import component styles
import {
  SliderContainer,
  SliderTitle,
  Card,
  Image,
  Icons,
  Icon,
  Info,
  Description,
  Category,
  Weigth,
  Title,
  Price,
  GoCart,
  Tooltip,
} from "./styles";

// install Swiper modules
SwiperCore.use([Navigation, Pagination, Scrollbar, A11y]);



const Slider = () => {
  
  // API to get products
  const API = `https://superfuds-assets.s3-sa-east-1.amazonaws.com/utils/product.json`;
  // state to store the products
  const [product, setProduct] = useState([]);
  // use the context app
  const {state, addToCart} = useContext(AppContext)
  
  // get products
  useEffect(() => {
    fetch(API)
    .then((response) => response.json())
    .then((data) => setProduct(data));
  }, []);
  
  // function for add products to cart
  const handleAddToCart = (product) => () => {
    addToCart(product);
  };

  return (
    <SliderContainer>
      <SliderTitle>
        <h2>Nuevo en superFuds</h2>
        <a href="" target="_blank" rel="noopener noreferrer">ver mas</a>
      </SliderTitle>
      <Swiper
        spaceBetween={30}
        breakpoints={
           {
             320: {
              slidesPerView: 1.2,
              spaceBetween: 20
            },
             768: {
              slidesPerView: 3.3,
              spaceBetween: 20
            },
             992: {
              slidesPerView: 4.3,
              spaceBetween: 20
            },
             1200: {
              slidesPerView: 5.2,
              spaceBetween: 20
            },
            }
        }
        navigation={{
          prevEl: ".prev",
          nextEl: ".next",
        }}
        loop
      >
        {product.map((item) => (
          <SwiperSlide key={item.id}>
            <Card>
              <Image>
                <img src={item.thumbnail} alt="" />
                <Icons>
                {
                  item.sellos.map((sello, index)=>(
                  <Icon key={index}>
                    <img src={sello.image} alt="" />
                    <Tooltip className="tooltip">
                      <h3>producto</h3>
                      <p>{sello.name}</p>
                    </Tooltip>
                  </Icon>
                  ))
                }
                </Icons>
              </Image>
              <Info>
                <Description>
                  <Category>{item.category}</Category>
                  <Weigth>{item.net_content}</Weigth>
                </Description>
                <Title>{item.title}</Title>
                <Price>
                  <span>$</span>
                  {item.price_real}
                  <small> x {item.units_sf} unids</small>
                </Price>
              </Info>
              <GoCart className={"btn"} onClick={handleAddToCart(item)}>agregar al carrito</GoCart>
            </Card>
          </SwiperSlide>
        ))}
      </Swiper>
      <div className="swiper-button-prev prev" />
      <div className="swiper-button-next next" />
    </SliderContainer>
  );
};

export default Slider;