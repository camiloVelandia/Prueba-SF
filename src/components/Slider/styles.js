import styled from "styled-components";


export const SliderContainer = styled.section`
  max-width: 1330px;
  padding-left: 38px;
  padding-right: 38px;
  margin: 0 auto;
  margin-top: 90px;
  position: relative;

  & .next::after {
    color: var(--brand);
    font-size: 2.4rem;
    font-weight: 700;
  }

  & > .prev::after {
    color: var(--brand);
    font-size: 2.4rem;
    font-weight: 700;
  }

  @media (min-width: 1200px) {
    margin-top: 150px;
    padding-left: 80px;
    padding-right: 80px;
  }
`;


export const SliderTitle = styled.div`
  display:flex;
  align-items:center;

  & h2{
    font-size:1.9rem;
    margin:0;
  }

  & a{
    text-decoration:none;
    color:var(--brand);
    font-size:1.5rem;
    font-weight:700;
    padding-left: 20px;
  }
`;


export const Card = styled.div`
  box-shadow: -1px 0px 7px 0px rgb(0 0 0 / 34%);
  padding: 5px;
  margin: 20px 10px;
  border-radius:8px;
  &:hover {
    .btn {
      display: block;
    }
  }
`;


export const Image = styled.figure`
  height:150px;

  & > img{
    width:100%;
    height:100%;
    object-fit:contain;
  }
`;


export const Icons = styled.div`
  position:absolute;
  top:40px;
  right:20px;
`;


export const Icon = styled.figure`
  width:20px;
  margin:0;
  margin-bottom:5px;
  cursor: pointer;
  position:relative;

  & img{
    width:100%;
    object-fit:contain;
  }

  &:hover{
    .tooltip{
      display:block;
    }
  }
`;


export const Tooltip = styled.div`
  background-color: #0e984c;
  position: absolute;
  top: -150%;
  left: 30px;
  padding: 8px;
  border-radius: 8px;
  display:none;
  z-index:9;
  &::before {
    content: "";
    width: 0;
    height: 0;
    border-right: 16px solid #0e984c;
    border-top: 8px solid transparent;
    border-bottom: 8px solid transparent;
    position:absolute;
    right:94%;
    bottom:0;
  }

  & h3 {
    color: var(--yellow);
    font-size: 1.5rem;
    margin: 0;
    padding-bottom: 3px;
  }

  & p {
    margin: 0;
    color: var(--white);
    font-size: 1.5rem;
  }
`;


export const Info = styled.div`
  border-top: 1px solid #e0e0e0;
  padding-top: 5px;
`;


export const Description = styled.div`
  display:flex;
  justify-content:space-between;
`;


export const Category = styled.p`
  color:var(--brand);
  font-size:1.2rem;
  margin:0;
`;


export const Weigth = styled.p`
  font-size:1.2rem;
  background-color:var(--brand);
  color:var(--white);
  padding:2px 5px;
  border-radius:10px;
  margin:0;
`;


export const Title = styled.h2`
  font-size: 1.8rem;
  margin-bottom: 10px;
  text-overflow: ellipsis;
  white-space: nowrap;
  overflow: hidden;
`;


export const Price = styled.p`
  font-size:2rem;
  margin-bottom:10px;
  margin-top:0;
  font-weight:700;

  & span{
    color:var(--brand);
  }
  
  & small{
    color:var(--grey);
    font-size:1.3rem;
    font-weight:400
  }
`;


export const GoCart = styled.button`
  background-color: var(--brand);
  color: var(--white);
  width:100%;
  border:none;
  cursor: pointer;
  text-decoration:none;
  display:block;
  text-align:center;
  padding:10px;
  display:none;
  font-size:1.5rem;
  top:100%;
  left:0;
  right:0;
`;
