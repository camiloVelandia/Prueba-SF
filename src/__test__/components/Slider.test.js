import React from "react";
import { mount } from "enzyme";
import { create } from "react-test-renderer";
import Slider from "../../components/Slider";

describe("<Slider/>", () => {
  const slider = mount(<Slider />);
  test("render del componente Slider", () => {
    expect(slider.length).toEqual(1);
  });
});

describe("Slider snaaapshot", () => {
  test("Comprobar ui del Slider", () => {
    const slider = create(<Slider />);
    expect(slider.toJSON()).toMatchSnapshot();
  });
});
