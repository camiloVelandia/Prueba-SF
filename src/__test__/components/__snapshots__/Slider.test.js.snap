// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Slider snaaapshot Comprobar ui del Slider 1`] = `
<section
  className="sc-bdvvaa eiQRHP"
>
  <div
    className="sc-gsDJrp izHZSc"
  >
    <h2>
      Nuevo en superFuds
    </h2>
    <a
      href=""
      rel="noopener noreferrer"
      target="_blank"
    >
      ver mas
    </a>
  </div>
  <div
    className="swiper-container"
  >
    <div
      className="swiper-wrapper"
    />
  </div>
  <div
    className="swiper-button-prev prev"
  />
  <div
    className="swiper-button-next next"
  />
</section>
`;
